app/src/org/commcare/activities/EntityDetailActivity.java

```diff
@@ -251,4 +263,42 @@ public void onBackPressed() {
         super.onBackPressed();
         GoogleAnalyticsUtils.reportEntityDetailExit(false, mDetailView.getTabCount() == 1);
     }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        super.onCreateOptionsMenu(menu);
+        menu.add(0, MENU_PRINT_DETAIL, 0, Localization.get("menu.print.detail"));
+        return true;
+    }
+
+    @Override
+    public boolean onPrepareOptionsMenu(Menu menu) {
+        super.onPrepareOptionsMenu(menu);
+        menu.findItem(MENU_PRINT_DETAIL).setVisible(detail.isPrintEnabled());
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch(item.getItemId()) {
+            case MENU_PRINT_DETAIL:
+                printDetail();
+                return true;
+        }
+        return super.onOptionsItemSelected(item);
+    }
+
+    private void printDetail() {
+        Intent i = new Intent(this, TemplatePrinterActivity.class);
+        i.putExtra(TemplatePrinterActivity.PRINT_TEMPLATE_REF_STRING,
+                detail.getDerivedPrintTemplatePath());

```

Comment:
Yep. mLabour was very clear that the only thing they were interested in was being able to print the entire detail at once, so I just went with implementing that for now.